charts/spire/templates/spire-controller-manager-deployment.yaml

```diff
@@ -0,0 +1,71 @@
+{{- if eq (.Values.controllerManager.enabled | toString) "true" }}
+{{- $fullname := include "spire.fullname" . }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ printf "%s-controller-manager" $fullname }}
+  labels:
+    {{- include "spire.controllerManager.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.controllerManager.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "spire.controllerManager.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.controllerManager.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "spire.controllerManager.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- range . }}
+        - name: {{ printf "%s-%s" $fullname .name }}
+        {{- end }}
+      {{- end }}
+      serviceAccountName: {{ include "spire.serviceAccountName" . }}-agent
```

Comment:
Should we create a dedicated service account and RBAC?

Reply:
I'm not sure about that. It seems that it can cause duplicate the same CR/CRBs for the server for the controller manager too 🤔 Because I really don't know what are the exact permissions we need to have for controller manager